dm/master/shardddl/optimist.go

```diff
@@ -483,6 +483,17 @@ func (o *Optimist) handleLock(info optimism.Info, tts []optimism.TargetTable, sk
 	} else {
 		o.logger.Info("the shard DDL lock returned some DDLs",
 			zap.String("lock", lockID), zap.Strings("ddls", newDDLs), zap.Stringer("info", info), zap.Bool("is deleted", info.IsDeleted))
+
+		// try to record the init schema before applied the DDL to the downstream.
+		initSchema := optimism.NewInitSchema(info.Task, info.DownSchema, info.DownTable, info.TableInfoBefore)
+		rev, putted, err2 := optimism.PutInitSchemaIfNotExist(o.cli, initSchema)
+		if err2 != nil {
+			o.logger.Warn("fail to record the initial schema", zap.Stringer("info", info), log.ShortError(err2))
```

Comment:
do we need to return err2?

Reply:
changed to return `err2` in https://github.com/pingcap/dm/pull/726/commits/053fb6fb6b39c89e22a2ca30189abc4aead522b5.